### PR TITLE
Rename MAPO image

### PR DESCRIPTION
**Pull request by mandre**
The MAPO image is renamed from `machine-api-provider-openstack` to
`openstack-machine-api-provider` for consistency.

### CODE REVIEW

File: core-services/image-mirroring/openshift/mapping_origin_4_10 [hunk 1]
```diff
@@ -219,6 +218,7 @@ registry.ci.openshift.org/origin/4.10:openstack-cinder-csi-driver-operator quay.
 registry.ci.openshift.org/origin/4.10:openstack-cinder-csi-driver-operator-test quay.io/openshift/origin-openstack-cinder-csi-driver-operator-test:4.10 quay.io/openshift/origin-openstack-cinder-csi-driver-operator-test:4.10.0 quay.io/openshift/origin-openstack-cinder-csi-driver-operator-test:latest
 registry.ci.openshift.org/origin/4.10:openstack-cloud-controller-manager quay.io/openshift/origin-openstack-cloud-controller-manager:4.10 quay.io/openshift/origin-openstack-cloud-controller-manager:4.10.0 quay.io/openshift/origin-openstack-cloud-controller-manager:latest
 registry.ci.openshift.org/origin/4.10:openstack-installer quay.io/openshift/origin-openstack-installer:4.10 quay.io/openshift/origin-openstack-installer:4.10.0 quay.io/openshift/origin-openstack-installer:latest
+registry.ci.openshift.org/origin/4.10:openstack-machine-api-provider quay.io/openshift/origin-openstack-machine-api-provider:4.10 quay.io/openshift/origin-openstack-machine-api-provider:4.10.0 quay.io/openshift/origin-openstack-machine-api-provider:latest
```
Comment (smarterclayton) [inline core-services/image-mirroring/openshift/mapping_origin_4_10:221]: Is anyone depending on the old image name in an operator image-reference yet?  Is there a PR that renames it that you can link here (if so)?

Comment (mandre) [reply core-services/image-mirroring/openshift/mapping_origin_4_10:221]: Sure, here's the MAO patch to move to the new image https://github.com/openshift/machine-api-operator/pull/978
It's the only consumer of this image.

**Pull request merged**